### ISSUE TITLE
Unique constraint error on new post in board after moving threads across boards

### ISSUE THREAD

**Issue by divSelector** [divSelector/divchan#8]
It happens rarely and I haven't figured out why because it sometimes corrects itself after error and it doesn't always happen. But it's probably something to do with the last post Hid adjustment code when moving threads

**Comment by divSelector** [comment]
I'm going to be honest. I don't fully know why this happens, but this should fix it. Hopefully we're not running into infinite loops here in the future. :fearful: 

https://github.com/divSelector/divchan/commit/31758115d1a4ab4975f89fcc8ea4c19e4f9bddbd

**Comment by divSelector** [comment]
This was not nearly as simple as I wanted to believe and the above fix was awful.

It took me a while but I figured out what the issue really is.

There are board fixtures and thread fixtures and posts fixtures. The boards are initialized by the fixtures. If there are no threads or posts, the first ones will always start at lowest id and increment each time a post is made.

If posts and threads are loaded via fixtures rather than made through the app, the next post id will never increment to the correct index.

This why we are always starting at the rock index, and looping through every post index while trying and failing until one works is ridiculous. You can tell I kind knew that from the post post above.

TLDR what Im gonna do is move the `board.last_post_hid` increment logic to a pre_save signal and it will happen every time a post is saved.

**Comment by divSelector** [comment]
real fix

https://github.com/divSelector/divchan/commit/b264de2ce66ccf7a9f5593c971821bbc8e6f3a14

**Comment by divSelector** [comment]
kekw
